pkg/manifest/os.go

```diff
@@ -863,9 +868,51 @@ func (p *OS) serialize() osbuild.Pipeline {
 		}
 	}
 
+	if len(p.CACerts) > 0 {
+		for _, cc := range p.CACerts {
+			for _, c := range parseCerts(cc) {
+				path := filepath.Join("/etc/pki/ca-trust/source/anchors", filepath.Base(c.SerialNumber.Text(16))+".pem")
+				f, err := fsnode.NewFile(path, nil, "root", "root", pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: c.Raw}))
+				if err != nil {
+					panic(err)
+				}
+				pipeline.AddStages(osbuild.GenFileNodesStages([]*fsnode.File{f})...)
+			}
+		}
+		pipeline.AddStage(osbuild.NewCAStageStage())
+	}
+
 	return pipeline
 }
 
+func parseCerts(cert string) []*x509.Certificate {
+	result := make([]*x509.Certificate, 0, 1)
+	block := []byte(cert)
+	var blocks [][]byte
+	for {
+		var certDERBlock *pem.Block
+		certDERBlock, block = pem.Decode(block)
+		if certDERBlock == nil {
+			break
+		}
+
+		if certDERBlock.Type == "CERTIFICATE" {
+			blocks = append(blocks, certDERBlock.Bytes)
+		}
+	}
+
+	for _, block := range blocks {
+		cert, err := x509.ParseCertificate(block)
+		if err != nil {
+			logrus.Warnf("failed to parse certificate, skipping: %s", err)
```

Comment:
Is this the behavior we want/expect? This will mean that pasing an invalid cert string will just result in an empty list and the cert not appearing in the manifest. As a user I would probably be surprised if I paste a cert with an error and then get an image (much later) that does not contain it. Sorry, I think this is a very wordy way of saysing: "it seems to me we should return an error here" (unless I miss something of course which is entriely possible)

Reply:
Good call, explicit error it is.